api/kroxylicious-api/src/main/java/io/kroxylicious/proxy/service/Contributor.java

```diff
@@ -28,9 +27,9 @@
      * Creates an instance of the service.
      *
      * @param shortName service short name
-     * @param proxyConfig proxy configuration
+     * @param endpointProvider endpoint provider
      * @param config service configuration which may be null if the service instance does not accept configuration.
      * @return the service instance
      */
-    T getInstance(String shortName, ProxyConfig proxyConfig, BaseConfig config);
+    T getInstance(String shortName, ClusterEndpointProvider endpointProvider, BaseConfig config);
```

Comment:
The `ClusterEndpointProvider` parameter should disappear.  At the moment the BrokerAddressFilter needs it to process the broker addresses in the RPC response.  As soon as Kroxy takes responsibility for this task, this FP can be removed.